ibl_pipeline/analyses/behavior.py

```diff
@@ -6,6 +6,69 @@
 schema = dj.schema(dj.config.get('database.prefix', '') + 'ibl_analyses_behavior')
 
 
+@schema
+class Contrasts(dj.Computed):
+    definition = """
+    -> behavior.TrialSet
+    ---
+    contrasts:                      blob    # contrasts used in this session, negative when on the left
+    has_no_contrast:                bool    # have trials with no contrast
+    has_left:                       bool    # have trials with contrast on the left
+    has_right:                      bool    # have trials with contrast on the right
+    contrasts_left=null:            blob    # contrasts shown on the left
+    contrasts_right=null:           blob    # contrasts shown on the right
+    n_trials_stim:                  blob    # number of trials for each contrast
+    n_trials_stim_left=null:        blob    # number of trials when the stimulus is shown on the left
+    n_trials_stim_right=null:       blob    # number of trials when the stimulus is shown on the right
+    n_trials_stim_no_contrast=null: int     # number of trials when no stimulus is shown 
+    """
+
+    def make(self, key):
+        print(key)
+        key_con = key.copy()
+        trials = behavior.TrialSet.Trial & key
+        contrasts_left = np.unique(trials.fetch('trial_stim_contrast_left'))[1:] # discard contrast 0
```

Comment:
If you fetch from the same expression, it's better to do it as one query to save network delays:
```python
contrast_left, contrast_right = trials.fetch('trial_stim_contrast_left', 'trial_stim_contrast_right')
contrast_left, contrast_right = np.unique(contrast_left), np.unique(contrast_right)
```